modules/gg-fleet-statusd/src/fleet_status_service.c

```diff
@@ -281,43 +273,41 @@ GglError publish_fleet_status_update(
     }
 
     // check for a persisted sequence number
-    GglObject sequence;
+    GglObject sequence_obj;
     ret = ggl_gg_config_read(
         GGL_BUF_LIST(GGL_STR("system"), GGL_STR("fleetStatusSequenceNum")),
         &balloc.alloc,
-        &sequence
+        &sequence_obj
     );
-    if (ret == GGL_ERR_OK && sequence.type == GGL_TYPE_I64) {
+    int64_t sequence = 1;
+    if ((ret != GGL_ERR_OK) || (ggl_obj_type(sequence_obj) != GGL_TYPE_I64)) {
```

Comment:
This if-conditional is inverted.
```suggestion
    if ((ret == GGL_ERR_OK) && (ggl_obj_type(sequence_obj) == GGL_TYPE_I64)) {
```